core/src/trezor/loop.py

```diff
@@ -47,14 +47,22 @@
 
 
 def schedule(
-    task: Task, value: Any = None, deadline: int = None, finalizer: Finalizer = None
+    task: Task,
+    value: Any = None,
+    deadline: int = None,
+    finalizer: Finalizer = None,
+    reschedule: bool = False,
 ) -> None:
     """
     Schedule task to be executed with `value` on given `deadline` (in
     microseconds).  Does not start the event loop itself, see `run`.
     Usually done in very low-level cases, see `race` for more user-friendly
     and correct concept.
+
+    If `reschedule` is set, updates an existing entry.
     """
+    if reschedule:
+        _queue.discard(task)
```

Comment:
I haven't given this much thought, but I wonder, would it make sense to schedule one task multiple times? What happens if we do? Maybe schedule() should search the queue for the task, if it finds that it's already scheduled, then reschedule it. In other words maybe there is no need for the reschedule flag.

Reply:
`discard()` is a linear search, and `schedule()` is called relatively often.
I'm not sure how expensive this actually is on real hardware, but I was working off the assumption that this is the place for micro-optimizations ;)

Reply:
Re multiple scheduled tasks, i would consider that a new feature/requirement.
Currently we don't need it, and the queue would indeed be more correct if it explicitly disallowed that. There are some interesting usecases that could make use of a task scheduled multiple times, but then we have a problem with maintaining the state of the task object.

The main reason the queue does not reject that, AFAICT, is performance.

Reply:
Re "what happens": tasks are running coroutines/generators. You can safely schedule a task multiple times if, e.g., there is an infinite loop running inside it.
(a finished generator will simply die off, this is handled properly and does not cause a crash)
A complication is that the syscall machinery makes sure that the task is rescheduled whenever it awaits something. So e.g., if you schedule a task twice, it will run twice as often - and more importantly, the return values from the awaited syscalls will get intermingled.

In short, "bad things" is what would happen.
For that reason `loop.schedule` should generally not be used by application code.